micov/_io.py

```diff
@@ -304,10 +355,16 @@ def compress_from_stream(sam, bufsize=1_000_000_000, disable_compression=False):
         if len(buf) == 0:
             return None
 
-        if len(buf[0].split(b'\t')) == 3:
-            parse_f = parse_bed_cov_to_df
+        if isinstance(buf[0], str):
+            if len(buf[0].split('\t')) == 3:
+                parse_f = parse_bed_cov_to_df
+            else:
+                parse_f = parse_sam_to_df
         else:
-            parse_f = parse_sam_to_df
+            if len(buf[0].split(b'\t')) == 3:
+                parse_f = parse_bed_cov_to_df
+            else:
+                parse_f = parse_sam_to_df
```

Comment:
Interface wont allow applying suggestions over this block. But, what about this instead? It reduces replication and separates the concern of what the type of the buffer entry is from what parser to use

```python
        line = buf[0]
        if isinstance(line, str):
            delim = '\t'
        elif isinstance(line, bytes):
            delim = b'\t'
        else:
            raise ValueError(f"Unexpected buffer type: {type(line)}")

        if len(line.split(delim)) == 3:
            parse_f = parse_bed_cov_to_df
        else:
            parse_f = parse_sam_to_df
````

Reply:
This is much better. Thanks!